Fix ARG_MAX compilation error on libc-2.8, refer to:

http://bugs.gentoo.org/show_bug.cgi?id=225409

For more information.

Signed-off-by: Luiz Fernando N. Capitulino <lcapitulino@mandriva.com.br>

--- ./mono/io-layer/wapi_glob.c.orig	2008-06-13 09:27:08.361391740 +0200
+++ ./mono/io-layer/wapi_glob.c	2008-06-13 09:27:36.338060353 +0200
@@ -52,6 +52,13 @@
 #include <string.h>
 #include <unistd.h>
 
+#if defined(_SC_ARG_MAX)
+# if defined(ARG_MAX)
+#    undef ARG_MAX
+# endif
+# define ARG_MAX sysconf (_SC_ARG_MAX)
+#endif"
+
 #include "wapi_glob.h"
 
 #define	EOS		'\0'
